daemon/config_unix.go

```diff
@@ -82,6 +85,37 @@ func (config *Config) InstallFlags(cmd *flag.FlagSet, usageFn func(string) strin
 	cmd.StringVar(&config.CgroupParent, []string{"-cgroup-parent"}, "", usageFn("Set parent cgroup for all containers"))
 	cmd.StringVar(&config.RemappedRoot, []string{"-userns-remap"}, "", usageFn("User/Group setting for user namespaces"))
 	cmd.StringVar(&config.ContainerdAddr, []string{"-containerd"}, "", usageFn("Path to containerd socket"))
+	config.Runtimes = make(map[string]runtime.Runtime)
+	cmd.Var(runconfigopts.NewNamedRuntimeOpt("runtimes", &config.Runtimes), []string{"-runtime"}, usageFn("Add an OCI runtime to the list of available runtimes"))
```

Comment:
Yes, I found that too this morning, fixing it and adding more tests. Thanks
:)

On Sun, May 29, 2016 at 8:52 PM zhangwei_cs <notifications@github.com>
wrote:

> In daemon/config_unix.go
> <https://github.com/docker/docker/pull/22983#discussion_r65021281>:
>
> > @@ -82,6 +85,37 @@ func (config *Config) InstallFlags(cmd *flag.FlagSet, usageFn func(string) strin
> >  	cmd.StringVar(&config.CgroupParent, []string{"-cgroup-parent"}, "", usageFn("Set parent cgroup for all containers"))
> >  	cmd.StringVar(&config.RemappedRoot, []string{"-userns-remap"}, "", usageFn("User/Group setting for user namespaces"))
> >  	cmd.StringVar(&config.ContainerdAddr, []string{"-containerd"}, "", usageFn("Path to containerd socket"))
> > +	config.Runtimes = make(map[string]runtime.Runtime)
> > +	cmd.Var(runconfigopts.NewNamedRuntimeOpt("runtimes", &config.Runtimes), []string{"-runtime"}, usageFn("Add an OCI runtime to the list of available runtimes"))
>
> I vote to --add-runtime, because containerd has a --runtime flag with
> different meaning 😄
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/docker/docker/pull/22983/files/da0548d8aa947a2cc61aada1a6490c4c87190a40..53662739e29cfbfd11231be27104226d9ba3d828#r65021281>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/APUiflXzX61bWfGbPBtUVhdu-PewrviLks5qGl7rgaJpZM4Imvu8>
> .
>